src/software/world/ball_state.cpp

```diff
@@ -6,6 +6,15 @@ BallState::BallState(const Point& position, const Vector& velocity,
 {
 }
 
+BallState::BallState(const TbotsProto::BallState& ball_state_proto)
+    : position_(Point(ball_state_proto.global_position().x_meters(),
+                      ball_state_proto.global_position().y_meters())),
+      velocity_(Vector(ball_state_proto.global_velocity().x_component_meters(),
+                       ball_state_proto.global_velocity().y_component_meters())),
+      height_(ball_state_proto.distance_from_ground())
```

Comment:
I was thinking that if we're going to pass in a protobuf into a class then we could make the internal representation the protobuf itself, i.e. BallState has a `TbotsProto::BallState ball_state_proto` field and the constructor initializer list can simply be `ball_state_proto(ball_state_proto)`. Then to get a protobuf from a BallState, we can call a getter to get the `ball_state_proto`. What do you think? It'd be the same idea for RobotState and Timestamp as well.

Reply:
What would be the benefit of having another property in the class which stores the `ball_state_proto`, if the proto holds the same properties as the `BallState` class? Wouldn't it be simpler to translate those values to the already existing `BallState` properties? Then, when the getters are called (eg. `BallState::position()`), we already have the translated `Point` to return.
And are you referring to this new property only for `BallState`, `RobotState`, and `Timestamp`, or for all of the new proto classes I've added (eg. `World`)?